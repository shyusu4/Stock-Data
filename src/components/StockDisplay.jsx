/* eslint-disable */
import React, { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import Form from 'react-bootstrap/Form';
import { dispatchSymbols } from '../redux/stockData';
import Stock from './Stock';

const StockDisplay = () => {
  const dispatch = useDispatch();
  const stocks = useSelector((state) => state.stockReducer.filteredSymbols || []);
  const [query, setQuery] = useState('');

  useEffect(() => {
    if (!stocks.length) {
      dispatch(dispatchSymbols());
    }
  });

  return (
    <>
      <div className="stats">
        <Form className="d-flex">
          <p className="stats-text">Stats by company</p>
          <Form.Control
            type="search"
            placeholder="Search by company name or symbol"
            className="searchbar"
            aria-label="Search"
            onChange={(e) => setQuery(e.target.value)}
          />
        </Form>
      </div>
      <div className="stock-container">
        {
            stocks.filter((stock) => {
              if (query === '') {
                return stock;
              } if (stock.name.toLowerCase().includes(query.toLowerCase()) || stock.symbol.toLowerCase().includes(query.toLowerCase())) {
                return stock;
              }
            }).map((stock, index) => (
              <div className="stock-item" key={index}>
                <Stock
                  key={stock.symbol}
                  symbol={stock.symbol}
                  name={stock.name}
                  price={stock.price}
                  exchange={stock.exchange}
                  exchangeShortName={stock.exchangeShortName}
                />
              </div>
            ))
}
      </div>
      
    </>
  );
};

export default StockDisplay;
