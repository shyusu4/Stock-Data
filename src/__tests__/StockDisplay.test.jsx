import React from 'react';
import Renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import StockDisplay from '../components/StockDisplay';

window.matchMedia = window.matchMedia || function () {
  return {
    matches: false,
    addListener() {},
    removeListener() {},
  };
};

describe('StockDisplay', () => {
  it('Check if StockDisplay component renders correctly', () => {
    const tree = Renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <StockDisplay />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
