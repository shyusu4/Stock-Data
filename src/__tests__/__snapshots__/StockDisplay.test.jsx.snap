// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StockDisplay Check if StockDisplay component renders correctly 1`] = `
Array [
  <div
    className="stats"
  >
    <form
      className="d-flex"
    >
      <p
        className="stats-text"
      >
        Stats by company
      </p>
      <input
        aria-label="Search"
        className="searchbar form-control"
        onChange={[Function]}
        placeholder="Search by company name or symbol"
        type="search"
      />
    </form>
  </div>,
  <div
    className="stock-container"
  />,
]
`;
